src/components/flowersPage/hooks/useFlowerData.ts

```diff
@@ -0,0 +1,34 @@
+import { useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { FlowerSchemaType } from '../FlowersPage.types.ts';
+import { FlowerSchema } from '../FlowersPage.schema.ts';
+import { flowerQueryKey } from '../../../Backend.constants.ts';
+
+export const useFlowerData = () => {
+  const getData = async (): Promise<FlowerSchemaType['items'] | undefined> => {
+    const response = await fetch('mock-api/flowers');
+    if (!response.ok) {
+      throw new Error('Failed to fetch data');
+    }
+
+    const data: unknown = await response.json();
+
+    const parsed = FlowerSchema.safeParse(data);
+
+    return parsed.data?.items;
+  };
+
+  const { data, isError, isPending } = useQuery({
+    queryKey: [flowerQueryKey],
+    queryFn: getData,
+  });
+
+  return useMemo(
```

Comment:
Im not sure about that optimization because we are base in API on data which is object and everytime reference(from useMemo array dependency) to that object is different.  But maybe in our case flowers mocks are same and it works.

Reply:
Can we please discuss it during our call? I'm not sure I understand and what perhaps should be changed🤔